src/datachain/lib/dc.py

```diff
@@ -765,9 +797,11 @@ def map(
         """
         udf_obj = self._udf_to_obj(Mapper, func, params, output, signal_map)
 
-        chain = self.add_signals(
-            udf_obj.to_udf_wrapper(),
-            **self._settings.to_dict(),
+        chain = self.clone(
```

Comment:
just to make sure - does it mean that we do an extra table copy now?

Reply:
No, for 2 reasons:
* `new_table` is ignored if you pass `query`.
* `clone()` never makes an actual copy anyway, it only sets an alias for the current query.